components/brave_shields/browser/ad_block_service.cc

```diff
@@ -69,28 +69,45 @@ bool AdBlockService::ShouldStartRequest(const GURL& url,
 }
 
 bool AdBlockService::Init() {
-   if (!GetDATFileData(DAT_FILE, buffer_)) {
+  base::Closure registered_callback =
+    base::Bind(&AdBlockService::OnComponentRegistered,
+               base::Unretained(this), kAdBlockPlusUpdaterId);
+  ReadyCallback ready_callback =
+    base::Bind(&AdBlockService::OnComponentReady,
+               base::Unretained(this), kAdBlockPlusUpdaterId);
+  brave::RegisterComponent(g_browser_process->component_updater(),
+      kAdBlockPlusUpdaterName, kAdBlockPlusUpdaterBase64PublicKey,
+      registered_callback, ready_callback);
```

Comment:
Pls handle all of this in the base, and then pass `kAdBlockPlusUpdaterName`, `kAdBlockPlusUpdaterBase64PublicKey`, and `kAdBlockPlusUpdaterId` to the base class.